lib/Sema/CSGen.cpp

```diff
@@ -1699,7 +1699,8 @@ namespace {
         if (!CS.getASTContext().isSwiftVersionAtLeast(6)) {
           auto paramTypeVar = CS.createTypeVariable(
               CS.getConstraintLocator(expr, ConstraintLocator::ApplyArgument),
-              TVO_CanBindToLValue | TVO_CanBindToInOut | TVO_CanBindToNoEscape);
+              TVO_CanBindToLValue | TVO_CanBindToInOut | TVO_CanBindToNoEscape |
+                  TVO_CanBindToPack);
```

Comment:
My understanding is that  TVO_CanBindToPack means it _only_ binds to a pack, but @hborla can correct me if I'm wrong. We'll need to add a new flag if we want this type variable to bind to packs and non-packs.

Reply:
However the type of the parameter list is always going to be a tuple no? Why would it be a pack type?

Reply:
@slavapestov all of the `CanBind*` mean that type variable is allowed to bind to a pack but not restricted to it, just like `TVO_CanBindToLValue` for example.

> However the type of the parameter list is always going to be a tuple no? Why would it be a pack type?

No, unlabeled single parameter is represented by a paren type that's why this fix in necessary, I mentioned that explicitly in the comment.

Reply:
Previously, `TVO_CanBindToPack` was used as if it meant "must bind to pack", but I'm not sure there are any places in the code that rely on this assumption; most places that check `canBindToPack` are to delay constraint solving until the type variable is found. If the tests pass I think it's okay 👍

Reply:
From all the uses I've seen it makes more sense to have it as "can" too because it gives diagnostics a chance to assign other types instead of placeholders.